sonic_platform_base/sonic_xcvr/xcvr_api_factory.py

```diff
@@ -38,6 +39,9 @@ def create_xcvr_api(self):
             mem_map = CmisMemMap(codes)
             xcvr_eeprom = XcvrEeprom(self.reader, self.writer, mem_map)
             api = CmisApi(xcvr_eeprom)
+            if api.is_coherent_module():
```

Comment:
please verify the following case.

User changes the sfp module type by removing the older module and inserting the newer one when Xcvrd is running:

1. Port had QSFP-DD non-coherent optic and user inserts QSFP-DD coherent optic
2. Port had QSFP-DD coherent optic and user inserts QSFP-DD non-coherent optic
3. Port had no optics and user inserts a QSFP-DD coherent/non-coherent optic

Ensure for all above cases the API generated by xcvr_api_factory.py matches with the optics present in the port.

Reply:
will do the test

Reply:
Verified and all tests passed.